A51-xds-custom-backend-metrics.md

```diff
@@ -0,0 +1,382 @@
+As with per-request data, the protobuf deserialization will happen only once.  
+The data provided to the callback should be exposed to the LB policy using the same type as used for per-request data.
+
+The API for subscribing to out-of-band reports must be available to LB policies at any level of the routing hierarchy, 
+including both leaf LB policies (whose children are subchannels) and parent LB policies (whose children are other LB policies).  
+Note that there may be multiple LB policies in the routing hierarchy that are interested in subscribing.  
+For example, in the case where we are using the xDS policy for cluster routing and the WRR policy for intra-cluster routing,
+both policies may be interested in subscribing.  
+The WRR policy will be directly managing the subchannels and will therefore be able to subscribe directly against the subchannel API, 
+whereas the xDS policy will need to subscribe against the LB policy API of the WRR policy, 
+which will need to forward reports from the subchannels to its external subscriber(s) (i.e., the xDS policy).  
+Reports are forwarded as-is with no aggregation; each LB policy is responsible for performing whatever aggregation is appropriate for its use.
+
+Note that the LB policy API should be designed such that individual LB policies that do not know or 
+care about out-of-band reporting should not have to do anything special to support them.  
+For example, it should be possible for the xDS policy to subscribe to out-of-band reports from 
+a child policy like RR, which does not know anything about out-of-band-reports and will not contain any
+code to handle them.
+
+When a subchannel first establishes a connection, if at least one LB policy is subscribing to out-of-band reporting, 
+the client will immediately start the StreamCoreMetrics call to the backend.  
+The client does not need to wait for this call to be established before it starts sending other calls on the subchannel.
+
+If the StreamCoreMetrics call fails with status `UNIMPLEMENTED`, the client will not retry the call, 
+and the subscribing LB polic(ies) will not receive any callbacks.  
+However, the client will record a channel trace event indicating that this has happened.  
+It will also log a message at priority ERROR.
+
+If the StreamCoreMetrics call returns any other status, the client will retry the call.  
+To avoid hammering a server that may be experiencing problems, the client will use exponential backoff between attempts.  
+When the client receives a message from the server on a given call, the backoff state is reset, 
+so the next attempt will occur immediately, but any subsequent attempt will be subject to exponential backoff.
+
+When the client subchannel is shutting down or when the backend sends a GOAWAY, 
+the client will cancel the StreamCoreMetrics call.  
+There is no need to wait for the final status from the server in this case.
+
+## Use Cases
```

Comment:
I think this section probably isn't needed.  We cover the use-cases sufficiently well above.